tests/utils/runner/runner.go

```diff
@@ -0,0 +1,377 @@
+// Note: currently assumes one blockchain per subnet
+type Subnet struct {
+	// SubnetID is the txID of the transaction that created the subnet
+	SubnetID ids.ID
+	// Current ANR assumes one blockchain per subnet, so we have a single blockchainID here
+	BlockchainID ids.ID
+	// ValidatorURIs is the base URIs for each participant of the Subnet
+	ValidatorURIs []string
+}
+
+type ANRConfig struct {
+	LogLevel            string
+	AvalancheGoExecPath string
+	PluginDir           string
+	GlobalNodeConfig    string
+}
+
+// NetworkManager is a wrapper around the ANR to simplify the setup and teardown code
+// of tests that rely on the ANR.
+type NetworkManager struct {
+	ANRConfig ANRConfig
+
+	subnets []*Subnet
+
+	logFactory      logging.Factory
+	anrClient       runner_sdk.Client
+	anrServer       runner_server.Server
+	done            chan struct{}
+	serverCtxCancel context.CancelFunc
+}
+
+func NewDefaultANRConfig() ANRConfig {
+	defaultConfig := ANRConfig{
+		LogLevel:            "info",
+		AvalancheGoExecPath: os.ExpandEnv("$GOPATH/src/github.com/ava-labs/avalanchego/build/avalanchego"),
+		PluginDir:           os.ExpandEnv("$GOPATH/src/github.com/ava-labs/avalanchego/build/plugins"),
+		GlobalNodeConfig: `{
+			"log-display-level":"info",
+			"proposervm-use-current-height":true,
+			"throttler-inbound-validator-alloc-size":"107374182",
+			"throttler-inbound-node-max-processing-msgs":"100000",
+			"throttler-inbound-bandwidth-refill-rate":"1073741824",
+			"throttler-inbound-bandwidth-max-burst-size":"1073741824",
+			"throttler-inbound-cpu-validator-alloc":"100000",
+			"throttler-inbound-disk-validator-alloc":"10737418240000",
+			"throttler-outbound-validator-alloc-size":"107374182"
+		}`,
+	}
+	// If AVALANCHEGO_BUILD_PATH is populated, override location set by GOPATH
+	if envBuildPath, exists := os.LookupEnv("AVALANCHEGO_BUILD_PATH"); exists {
+		defaultConfig.AvalancheGoExecPath = fmt.Sprintf("%s/avalanchego", envBuildPath)
+		defaultConfig.PluginDir = fmt.Sprintf("%s/plugins", envBuildPath)
+	}
+	return defaultConfig
+}
+
+// NewNetworkManager constructs a new instance of a network manager
+func NewNetworkManager(config ANRConfig) *NetworkManager {
+	manager := &NetworkManager{
+		ANRConfig: config,
+	}
+
+	logLevel, err := logging.ToLevel(config.LogLevel)
+	if err != nil {
+		panic(fmt.Errorf("invalid ANR log level: %w", err))
+	}
+	manager.logFactory = logging.NewFactory(logging.Config{
+		DisplayLevel: logLevel,
+		LogLevel:     logLevel,
+	})
+
+	return manager
+}
+
+// startServer starts a new ANR server and sets/overwrites the anrServer, done channel, and serverCtxCancel function.
+func (n *NetworkManager) startServer(ctx context.Context) (<-chan struct{}, error) {
+	done := make(chan struct{})
+	zapServerLog, err := n.logFactory.Make("server")
+	if err != nil {
+		return nil, fmt.Errorf("failed to make server log: %w", err)
+	}
+
+	n.anrServer, err = runner_server.New(
+		runner_server.Config{
+			Port:                ":12352",
```

Comment:
should these be a part of config?

Reply:
Ya I think it's fine for the port to be a part of the config. Also, this is using the avalanche-network-runner code, and I think we should use the current interface rather than try to push style changes to that dependency unless we're doing a larger re-write.

Overall, I added this here to try and keep the use of ANR as well-separated from our actual tests as possible.